tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs

```diff
@@ -761,12 +761,37 @@ public async Task GetAuthorizationRequestUrl_NoHint_NoCcsRoutingHint_TestAsync()
             }
         }
 
+        [TestMethod]
+        public async Task DoNotUseNullCcsRoutingHint_TestAsync()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                ConfidentialClientApplication app = CreateCca(httpManager);
+
+                var uri = await app
+                    .GetAuthorizationRequestUrl(TestConstants.s_scope)
+                    .WithCcsRoutingHint("", "")
+                    .ExecuteAsync(CancellationToken.None)
+                    .ConfigureAwait(false);
+
+                AssertCcsHint(uri, "");
+            }
+        }
+
         private static void AssertCcsHint(Uri uri, string ccsHint)
         {
             Assert.IsNotNull(uri);
             Dictionary<string, string> qp = CoreHelpers.ParseKeyValueList(uri.Query.Substring(1), '&', true, null);
-            Assert.IsTrue(qp.ContainsKey(Constants.CcsRoutingHintHeader));
-            Assert.AreEqual(ccsHint, qp[Constants.CcsRoutingHintHeader]);
+
+            if (!string.IsNullOrEmpty(ccsHint))
+            {
+                Assert.IsTrue(qp.ContainsKey(Constants.CcsRoutingHintHeader));
+                Assert.AreEqual(ccsHint, qp[Constants.CcsRoutingHintHeader]);
+            }
+            else
+            {
+                Assert.IsTrue(!qp.ContainsKey(Constants.CcsRoutingHintHeader));
```

Comment:
`Assert.IsFalse(qp.ContainsKey(Constants.CcsRoutingHintHeader));`?

Reply:
This is an assertion to verify that the CCS hint is not added if the WithCcs... api is given a null value